cls/SourceControl/Git/WebUIDriver.cls

```diff
@@ -344,6 +344,55 @@ ClassMethod HandleRequest(pagePath As %String, InternalName As %String = "", Out
                     }
                     set handled = 1
                 }
+            } elseif (pathStart = "git-command-unstage") {
+                set stringBody = ""
+                while '%request.Content.AtEnd {
+                    set stringBody = stringBody _ %request.Content.Read()
+                }
+                set stringBody = $zconvert(stringBody,"I","UTF8")
+                set requestBody = ##class(%Library.DynamicObject).%FromJSON(stringBody)
+
+                set flags = requestBody.flags
+                if (flags '= "") {
+                    set flaglist = $listfromstring(flags,",")
+                    for i=1:1:$listlength(flaglist) {
+                        set flag = $listget(flaglist,i)
+                        set args($increment(args)) = flag
+                    }
+                }
+                set message = requestBody.message
+                if (message '= "") {
+                    set args($increment(args)) = "-m"
+                    set args($increment(args)) = message
+                }
+                set details = requestBody.details
+                if (details '= "") {
+                    set args($increment(args)) = "-m"
+                    set args($increment(args)) = details
+                }
+                set command = requestBody.command
+                set files = ##class(%Library.DynamicObject).%FromJSON(requestBody.files)
+                set files = files.files
+
+                set returnCode = ##class(SourceControl.Git.Utils).RunGitCommandReStage(.outStream,.errStream,command,.files,args...)
+
+                set %data = ##class(%Stream.TmpCharacter).%New()
```

Comment:
Minor / tech debt: would be good to refactor out some of the copy/pasted code here (at the beginning and end).